test/open_api/map_test.exs

```diff
@@ -0,0 +1,403 @@
+                     "$ref": "#/components/parameters/ContentTypeHeader"
+                   }
+                 ],
+                 post: %OpenApiSpex.Operation{
+                   parameters: [
+                     %OpenApiSpex.Reference{},
+                     %OpenApiSpex.Reference{},
+                     %OpenApiSpex.Parameter{}
+                   ],
+                   deprecated: false,
+                   operationId: "example-post-test",
+                   requestBody: requestBody,
+                   callbacks: operationCallbacks,
+                   responses: operationResponses,
+                   security: operationSecurity,
+                   tags: ["test"],
+                   summary: "/example post summary",
+                   description: "/example post description",
+                   externalDocs: %OpenApiSpex.ExternalDocumentation{
+                     description: "Find more info here",
+                     url: "https://example.com"
+                   }
+                 }
+               }
+             } = paths
+
+      assert %{
+               petstore_auth: ["write:pets", "read:pets"]
+             } == operationSecurity
+
+      assert %{
+               "200": %OpenApiSpex.Response{}
+             } = operationResponses
+
+      assert %{
+               operationCallback: %{
+                 "http://server-a.com?transactionId={$request.body#/id}": %OpenApiSpex.PathItem{}
+               }
+             } = operationCallbacks
+
+      assert %OpenApiSpex.Schema{
+               properties: properties
+             } =
+               get_in(
+                 requestBody,
+                 [:content, :"application/json", :schema]
+                 |> Enum.map(&Access.key/1)
+               )
+
+      passengers =
+        get_in(
+          properties,
+          [:data, :properties, :passengers]
+          |> Enum.map(&Access.key/1)
+        )
+
+      assert %OpenApiSpex.Schema{} = passengers
+
+      assert %OpenApiSpex.Schema{
+               type: :string,
+               enum: ["adult", "child"]
+             } =
+               get_in(
+                 passengers,
+                 [:items, :properties, :type]
+                 |> Enum.map(&Access.key/1)
+               )
+
+      # ----------------
+
+      # TODO: Can move all of the code below into another test
+      # TODO: Can move the below into a setup fn
+      my_conn =
+        conn(:post, "/example?myParam=1", %{
+          "data" => %{
+            "first_name" => "Walter",
+            "given_name" => "Carvalho",
+            "phone_number" => "+447916844123"
+          }
+        })
+        |> put_req_header("content-type", "application/json")
+        |> put_req_header("accept-encoding", "gzip")
+
+      my_conn = fetch_query_params(my_conn)
+
+      assert {:ok, validation_result} =
+               OpenApiSpex.cast_and_validate(
+                 spec,
+                 spec.paths."/example".post,
```

Comment:
Yes, it is an atom here and we likely do not want to convert random atoms to strings, because of DoS, so we likely want to convert strings to atoms a bit more carefully. :)

Reply:
`paths` have string keys anyway: https://github.com/open-api-spex/open_api_spex/blob/master/lib/open_api_spex/paths.ex#L14